pkg/machinepool/helper_test.go

```diff
@@ -120,9 +126,123 @@ var _ = Describe("Machine pool helper", func() {
-			output := getSubnetFromUser(Cmd, r, true, mockClusterReady)
+			cmd.Flags().StringVar(&args.Subnet, "subnet", "", "")
+			cmd.Flags().Set("subnet", "test-subnet")
+			output := getSubnetFromUser(cmd, r, true, mockClusterReady, args)
 			Expect(output).To(Equal("test-subnet"))
 		})
 	})
 })
+
+var _ = Describe("getMachinePoolAvailabilityZones Functionality", func() {
+	var (
+		cluster    *cmv1.Cluster
+		r          *rosa.Runtime
+		mockClient *mock.MockClient
+		mockCtrl   *gomock.Controller
+	)
+
+	BeforeEach(func() {
+		r = &rosa.Runtime{}
+		mockCtrl = gomock.NewController(GinkgoT())
+		mockClient = mock.NewMockClient(mockCtrl)
+		r.AWSClient = mockClient
+
+		var err error
+		cluster = MockCluster(func(c *cmv1.ClusterBuilder) {
+			c.State(cmv1.ClusterStateReady)
+			b := cmv1.HypershiftBuilder{}
+			b.Enabled(true)
+			c.Hypershift(&b)
+			c.MultiAZ(true).Nodes(cmv1.NewClusterNodes().AvailabilityZones("us-east-1a", "us-east-1b"))
+		})
+		Expect(err).ToNot(HaveOccurred())
+		Expect(cluster.MultiAZ()).To(Equal(true))
+	})
+
+	Context("When testing getMachinePoolAvailabilityZones", func() {
+		It("Tests getMachinePoolAvailabilityZones with and without subnet input", func() {
+			multiAZMachinePool := false
+			availabilityZoneUserInput := "us-east-1a"
+			subnetUserInput := ""
+
+			azs, err := getMachinePoolAvailabilityZones(r, cluster,
+				multiAZMachinePool, availabilityZoneUserInput, subnetUserInput)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(azs).To(Equal([]string{"us-east-1a"}))
+
+			multiAZMachinePool = true
+			azs, err = getMachinePoolAvailabilityZones(r, cluster,
+				multiAZMachinePool, availabilityZoneUserInput, subnetUserInput)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(azs).To(Equal([]string{"us-east-1a", "us-east-1b"}))
+
+			// Test with subnet input
+			subnetUserInput = "subnet-123"
+			mockClient.EXPECT().GetSubnetAvailabilityZone(subnetUserInput).
+				Return("us-east-1a", nil)
+
+			azs, err = getMachinePoolAvailabilityZones(r, cluster,
+				multiAZMachinePool, availabilityZoneUserInput, subnetUserInput)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(azs).To(Equal([]string{"us-east-1a"}))
+		})
+	})
+})
+
+var _ = Describe("When testing getSubnetFromAvailabilityZone", func() {
+	var (
+		r              *rosa.Runtime
+		cmd            *cobra.Command
+		args           *MachinePoolArgs
+		mockClient     *mock.MockClient
+		az             string
+		subnetId1      string
+		subnetId2      string
+		privateSubnets []ec2types.Subnet
+		cluster        *cmv1.Cluster
+	)
+
+	BeforeEach(func() {
+		mockClient = mock.NewMockClient(gomock.NewController(GinkgoT()))
+		r = &rosa.Runtime{AWSClient: mockClient}
+		cmd = &cobra.Command{}
+		az = "us-east-1a"
+		subnetId1 = "subnet-123"
+		subnetId2 = "subnet-456"
+		args = &MachinePoolArgs{}
+	})
+
+	Context("When testing getSubnetFromAvailabilityZone", func() {
+		It("Tests getSubnetFromAvailabilityZone with no availability zone set and one subnet returned", func() {
```

Comment:
Can we flesh these test descriptions out a little, its kinda hard to understand what we are testing for
```suggestion
	       When("no availability zone is set", func() {
		          It("is should return available subnets", func() {
```